File: Java/3542. Minimum Operations to Convert All Elements to Zero.java
class Solution {
    public int minOperations(int[] nums) {
        int ans = 0;
        Stack<Integer> stack = new Stack<>();

        for (int n: nums) {
            while (!stack.isEmpty() && stack.peek() > n) {
                stack.pop();
            }
            if (stack.isEmpty() || stack.peek() < n) {
                if (n > 0) {
                    ans++;
                }
                stack.push(n);
            }
        }
        return ans;
    }
}

File: Python3/3542. Minimum Operations to Convert All Elements to Zero.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        ans = 0
        stack = [0]

        for n in nums:
            while stack and stack[-1] > n:
                stack.pop()

            if not stack or stack[-1] < n:
                if n > 0:
                    ans += 1
                stack.append(n)

        return ans